lib/fauxhai.rb

```diff
@@ -1,8 +1,8 @@
 module Fauxhai
-  require 'fauxhai/exception'
-  require 'fauxhai/fetcher'
-  require 'fauxhai/mocker'
-  require 'fauxhai/version'
+  autoload :Exception, 'fauxhai/exception'
```

Comment:
It might not matter too much right now, but it looks like Matz wants to deprecate autoload in Ruby 3.0. 

https://bugs.ruby-lang.org/issues/5653

Reply:
Which is super annoying for single threaded apps that want lower memory usage.

Reply:
I will deal with this when (if ever) it comes :) The ticket is very old, and I suspect it will be a long time still until this is a reality :) (mostly because Rails uses a fuckton of autoload to keep startup times down)